pkg/batcher/controllers/default.go

```diff
@@ -0,0 +1,307 @@
+/*
+Copyright 2019 kubeflow.org.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/go-logr/logr"
+	"io/ioutil"
+	"net/http"
+	"sync"
+	"time"
+	"errors"
+	"context"
+	"encoding/json"
+	"github.com/astaxie/beego"
+	"github.com/satori/go.uuid"
+)
+
+const (
+	SleepTime    = time.Microsecond * 100
+	MaxBatchsize = 32
+	MaxLatency   = 1.0
+)
+
+var (
+	channelIn = make(chan Input)
+	batcherInfo BatcherInfo
+	batcherHandler *BatcherHandler = nil
+	mutex sync.Mutex
+)
+
+type MainController struct {
+	beego.Controller
+}
+
+type Request struct {
+	Instances []interface{} `json:"instances"`
+}
+
+type Input struct {
+	ContextInput *context.Context `json:"contextInput"`
+	Instances *[]interface{} `json:"instances"`
+	ChannelOut *chan Response `json:"channelOut"`
+}
+
+type InputInfo struct {
+	ChannelOut *chan Response `json:"channelOut"`
+	Index []int `json:"index"`
+}
+
+type Response struct {
+	BatchID string `json:"batchId"`
+	Predictions []interface{} `json:"predictions"`
+}
+
+type ResponseError struct {
+	Message string `json:"message"`
+}
+
+type Predictions struct {
+	Predictions []interface{} `json:"predictions"`
+}
+
+type BatcherInfo struct {
```

Comment:
`BatcherInfo` is internal state, does it need to be json serializable ?

Reply:
It was used for debug. Shall I remove it?